app/src/main/kotlin/io/intrepid/skotlinton/screens/example2/Example2Presenter.kt

```diff
@@ -5,26 +5,23 @@ import io.intrepid.skotlinton.base.PresenterConfiguration
 import io.intrepid.skotlinton.utils.RxUtils
 import io.reactivex.functions.Consumer
 
-internal class Example2Presenter(view: Example2Contract.View, configuration: PresenterConfiguration)
-    : BasePresenter<Example2Contract.View>(view, configuration), Example2Contract.Presenter {
+class Example2Presenter(view: Example2Screen, configuration: PresenterConfiguration)
+    : BasePresenter<Example2Screen>(view, configuration) {
 
-    override fun onViewCreated() {
-        super.onViewCreated()
-
-        val disposable = restApi.getMyIp()
+    override fun screenCreated() {
+        disposables.add(restApi.getMyIp()
```

Comment:
in terms of formatting, it would be cleaner to separate this into two calls.

Reply:
How about a method that encapsulates and returns the underlying observable

Reply:
If we bring in [`RxKotlin`](https://github.com/ReactiveX/RxKotlin) (or just define our own extension method), we could change this to `disposables += ...` which would clean up the usage a bit.